packages/flutter_tools/lib/src/compile.dart

```diff
@@ -216,26 +214,26 @@ class KernelCompiler {
   final List<String> _fileSystemRoots;
   final StdoutHandler _stdoutHandler;
 
-  Future<CompilerOutput> compile({
-    String sdkRoot,
-    String mainPath,
-    String outputFilePath,
-    String depFilePath,
+  Future<CompilerOutput?> compile({
+    required String sdkRoot,
```

Comment:
This seems to actually be required, it's nullability isn't checked anywhere and it's always passed in.

Reply:
yes, definitely is